src/wallet/SavingsTransaction.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { FormattedMessage, FormattedRelative } from 'react-intl';
+import Avatar from '../components/Avatar';
+import { numberWithCommas } from '../helpers/regexHelpers';
+
+const getSavingsTransactionMessage = (transactionType, transactionDetails) => {
+  switch (transactionType) {
+    case 'cancel_transfer_from_savings':
+      return (
+        <FormattedMessage
+          id="cancel_transfer_from_savings"
+          defaultMessage="Cancel transfer from savings (request {requestId})"
+          values={{
+            requestId: transactionDetails.request_id,
+          }}
+        />
+      );
+    case 'transfer_to_savings':
+      return (
+        <FormattedMessage
+          id="transfer_to_savings"
+          defaultMessage="Transfer to savings {amount} to {username}"
+          values={{
+            amount: numberWithCommas(transactionDetails.amount),
+            username: <Link to={`/@${transactionDetails.to}`}>{transactionDetails.to}</Link>,
+          }}
+        />
+      );
+    case 'transfer_from_savings':
+    default:
```

Comment:
Is this intentional? If it is, are we sure that any other text will mean that this transfer was `transfer_from_savings`?

Reply:
yeah it was intentional, I figured since I filter by `savings` transactions anyways then one of them can be defaulted.. but I updated it now to default to return null